processor/src/lib.rs

```diff
@@ -586,16 +586,20 @@ impl Process {
         match decorator {
             Decorator::Advice(injector) => {
                 let advice_provider = host.advice_provider_mut();
```

Comment:
Once we get rid of `Decorator::Advice`, most of the code in this function will move into `Process::on_emit()`, right?

And there, we'll probably do the dispatch based on `event_id`, correct?

Reply:
yup it already is (in `Process::op_emit()`, we dispatch to `handle_advice_injector()` for appropriate `event_id`s